articles/anomaly-detection/passwords-checking.md

```diff
@@ -0,0 +1,229 @@
+come from security breached experienced by a third party application that have been released and so they might have been accessed by attackers. If a compromised account is detected different actions may be taken based on your configuration.
+
+On the other hand, bulk passwords checking aims to give you the power to audit your users security against your own
+passwords list, those passwords may come from different sources, examples of possible sources might be list of passwords
+available on the Internet (also known as passwords-dictionaries), your own hand-made checking list, etc.
+
+Combined with other tools we offer, actively auditing your users' account passwords will help you to proactively
+protect them and take risk mitigation actions in case a weak or compromised password gets detected.
+
+## Security considerations
+This tool helps you audit and protect your users against weak or compromised passwords, however, if used inappropriately, it might allow an attacker to brute force your users; because of this is really important for you to keep your credentials secure and not to issue tokens that allows to execute bulk passwords checking if not strictly necessary, also, make sure to include a token identifier and set a short expiration period on the tokens that allow this action, that way you will be able to blacklist them if they get compromised. Finally, delete the data once you have finished; that way you can be sure nobody else will be able to access it.
+
+We also take explicit measures to protect your data:
+- As everything in Auth0 all the data is transmitted over TLS even in our internal systems.
+- Your data will be in our systems as less time as possible, only the time needed to
+process and hand you in the results, then it is going to be deleted automatically in a short period of time. Even so,
+it is considered a good practice to manually delete the data once you have the results (see bellow).
+- Input data will be processed in batches and every batch will be deleted as soon as it gets processed to minimize the risk of
+exposure.
+- Result won't include the actual matches but a reference to them.
+
+To sum up:
+- Do not issue tokens that allows `create:passwords_checking_job` if you are not absolutely sure this right is needed
+- Set a short expiration on the issued tokens
+- Add an identifier (jti) to your tokens. So you will be able to blacklist compromised tokens.
+- If you think  a token might have been compromised blacklist it immediately.
+- Delete the input data and results once finished.
+
+## Usage
+This tool is made available through our [Management API](/api/management/v2). It has an endpoint that allows you to check a pre-defined list of users and passwords, which must be provided as a csv file, against a database connection. Each password will be checkend in background and some time later you will be able to get the results, the errors, and finally delete the information from Auth0 service.
```

Comment:
This tool is made available through our [Management API](/api/management/v2). It has an endpoint that allows you to check a pre-defined list of users and passwords --provided as a csv file-- against a database connection. This is run as a background process, which, when complete, will allow you to retrieve the results and any errors, as well as delete the data from Auth0.